ci/do_coverity_local.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+#
+#  do_coverity_local.sh
+#
+#  This script builds Envoy with the Coverity Scan Built Tool.
+#
+#  It expects the following environment variables to be set:
+#    COVERITY_TOKEN      - set to the user's Coverity Scan project token.
+#    COVERITY_USER_EMAIL - set to the email address used with the Coverity account.
+#                          defaults to the local git config user.email.
+
+
+set -e
+
+. ./ci/envoy_build_sha.sh
+
+[[ -z "${ENVOY_DOCKER_BUILD_DIR}" ]] && ENVOY_DOCKER_BUILD_DIR=/tmp/envoy-docker-build
+mkdir -p ${ENVOY_DOCKER_BUILD_DIR}
+
+TEST_TYPE="bazel.coverity"
+COVERITY_USER_EMAIL=${COVERITY_USER_EMAIL:-$(git config user.email)}
+COVERITY_OUTPUT_FILE="${ENVOY_DOCKER_BUILD_DIR}"/envoy/source/exe/envoy-coverity-output.tgz
+
+if [ -n "$COVERITY_TOKEN" ]
+then
+  pushd ${ENVOY_DOCKER_BUILD_DIR}
+  rm -rf cov-analysis
+  wget https://scan.coverity.com/download/linux64 --post-data "token=${COVERITY_TOKEN}&project=Envoy+Proxy" -O coverity_tool.tgz
+  tar xvf coverity_tool.tgz
+  mv cov-analysis-linux* cov-analysis
+  popd
+else
+  echo "ERROR: COVERITY_TOKEN is required to download and run Coverity Scan."
+  exit 1
+fi
+
+ci/run_envoy_docker.sh "ci/do_ci.sh ${TEST_TYPE}"
+
+if [ -z "$COVERITY_TOKEN" ]
```

Comment:
How can we get here? Above there is an else branch on `-n` which will `exit`.

Reply:
Can't get here. The first version of this PR had a COVERITY_SCAN_DIR that optionally allowed for specifying an alternative path to find Coverity instead of downloading it which the first check dealt with instead of being a conditional on COVERITY_TOKEN.
 I will clean this up.